### ISSUE TITLE
Windows py3.5 build is failing

### ISSUE THREAD

**Issue by jorisvandenbossche** [conda-forge/geopandas-feedstock#9]
See https://github.com/conda-forge/geopandas-feedstock/pull/8

It fails after the build when testing the packages:

```
BUILD END: geopandas-0.2-py35_0
TEST START: geopandas-0.2-py35_0
Fetching package metadata: ........ 
Solving package specifications: .... 
Error: The following specifications were found to be in conflict:
  - geopandas 0.2 py35_0
Use "conda info <package>" to see the dependencies for each package. 
Command exited with code 1
```

**Comment by jorisvandenbossche** [comment]
@ocefpaf Is there a way to try to debug this? (so I can help?) 
I looked at the dependencies, but they all seem to be building for py35

**Comment by ocefpaf** [comment]
We have a [script](https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/list_deps.py) to investigate dependencies, but sometime I just comment all dependencies and keep un-commenting them until I find the culprit.

PS: The reason I did not do it yet is b/c I will re-building everything that depends on `gdal` for `gdal 2.x` now that `fiona` supports it. So I will be re-building `geopandas` too and I will try to debug this.

**Comment by osuchw** [comment]
Is it possible that the culprit is jpeg?

_gdal_ demands:
` jpeg:      9b-vc14_0              conda-forge [vc14]`

where _matplotlib_ wants:
`jpeg:            8d-vc14_0         [vc14]`

**Comment by ocefpaf** [comment]
@osuchw that is the problem, but it is not `matplotlib`'s directly. It is one of its dependencies (`qt`).

```shell
> conda execute scripts/list_deps.py defaults --package qt --platform win-64 ; DONE                                                                                         ⏎
Fetching package metadata: ....
qt 4.8.7 vc10_4 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc10_5 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc10_6 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, zlib 1.2.*
qt 4.8.7 vc10_7 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc10_8 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc14_4 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc14_5 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc14_6 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, zlib 1.2.*
qt 4.8.7 vc14_7 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc14_8 win-64             : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc9_4 win-64              : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc9_5 win-64              : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc9_6 win-64              : jpeg 8d, libpng 1.6*, libtiff 4.0.*, zlib 1.2.*
qt 4.8.7 vc9_7 win-64              : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
qt 4.8.7 vc9_8 win-64              : jpeg 8d, libpng 1.6*, libtiff 4.0.*, openssl 1.0.2*, zlib 1.2.*
```

I am working on a solution in #10, but we need to build our own version of `qt` to make ti work.

**Comment by jjstarn-usgs** [comment]
Thanks so much for maintaining geopandas on conda forge! I am unable to install 0.2 for py3.5 on windows. Is this because of the issue you're discussing here, and is there any prognosis for a fix?

**Comment by ocefpaf** [comment]
Yeah. We are getting close to fix this though. There were a bunch of errors and bad builds (unrelated to `geopandas`) that needed resolving first.

**Comment by ocefpaf** [comment]
@jjstarn-usgs and @jorisvandenbossche this will probably be solved once we merge  https://github.com/conda-forge/staged-recipes/pull/1133 and https://github.com/conda-forge/staged-recipes/pull/1134

I appreciate if you could make some noise there to help me move this forward.

**Comment by fmaussion** [comment]
Hi all, my students are having the same problem. Any chance to get this working soon, or is a workaround available? I would appreciate any pointer. Thanks!

**Comment by ocefpaf** [comment]
The only workaround is to use Python 3.4 until conda-forge can package `Qt` for Windows. We are CI limited there b/c `Qt` build time is far more than `Appveyor`'s limit.

**Comment by fmaussion** [comment]
Alright, this is already more than enough. Thanks!

**Comment by jorisvandenbossche** [comment]
@ocefpaf the two PRs you mentioned above (for numexpr and pytables) are now merged. What is currently holding this up to have a working geopandas for Windows 3.5?

**Comment by ocefpaf** [comment]
The problem goes a little deeper. We need need `Qt` on Windows to be able to make all packages that depend on `matplotlib` work again on Windows. See https://github.com/conda-forge/matplotlib-feedstock/pull/72 and https://github.com/conda-forge/matplotlib-feedstock/issues/59

However, those are workarounds that will help PY27 and PY34. On PY35 we need to re-build `Qt` to make it consistent with conda-forge pinings. 

Because `Qt` builds takes longer than the minimum time `AppVeyor` grants us we need to build it locally and upload to the channel. Unfortunately that is not simple. Please give your support to https://github.com/conda-forge/conda-forge-enhancement-proposals/pull/5 to help us move forward and add `Qt` as soon as possible.

**Comment by jorisvandenbossche** [comment]
Can we for now just make a geopandas build here on conda-forge that does not depend on matplotlib (it's an optional dep anyway)? So at least people can install it (the shapely and fiona deps seems to have windows py3.5 builds)

(I can do that if that should work)

**Comment by ocefpaf** [comment]
> conda-forge that does not depend on matplotlib 

Sure. I thought that would not be desirable.

> I can do that if that should work)

I will modify the PR I already have. Hold on...

**Comment by ocefpaf** [comment]
@jorisvandenbossche I merged #10. Now we need to keep pushing for a `Qt` package on `conda-forge` to re-add `matplotlib` for `geopandas` by default.

**Comment by jorisvandenbossche** [comment]
@ocefpaf Thanks a lot!
